src/components/TodoList.vue

```diff
@@ -0,0 +1,50 @@
+<template>
+  <div>
+    <button id="add" @click="addEntry">Add</button>
+    <ol>
+      <TodoItem
+        v-for="(todo, index) in todoListData"
+        :todo="todo"
+        :index="index"
+        :key="todo.id"
+      />
+    </ol>
+  </div>
+</template>
+
+<script>
+import TodoItem from "./TodoItem";
+
+export default {
+  name: "TodoList",
+  components: { TodoItem },
+  data() {
+    return {
+      todoListData: [
+        { id: 1, message: "Foo", isEditing: false },
+        { id: 2, message: "Bar", isEditing: false },
+        { id: 3, message: "Baz", isEditing: false }
+      ],
+      lastID: 3
+    };
+  },
+  methods: {
+    addEntry: function() {
+      this.lastID++;
+      this.todoListData.push({
+        id: this.lastID,
```

Comment:
I have no idea what you mean by that?
Did you make the wrong code in your suggestion?

When saving we emit the index because it is much more efficient to search via index then having to look for the right id first.

Reply:
I think I made the suggestion before I learned that your `todo` object has a property `id`. In that case I would suggest to put all the things together

Am I right that you emit the todo along with a redundant id as another parameter?

Reply:
We changed it to emit todo object now (after the first code review)
Before we indeed each TodoItem received a Todo and an autogenerated Index.

when emiting a method we used the index and todo. so the Id was redundant.

But like I said we fixed that.